### PR TITLE
Homework week 4 Caglar

### PR DESCRIPTION
**Pull request by caglar-kaya**

### CODE REVIEW

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 1]
```diff
@@ -11,13 +11,12 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
...
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
+  const evenNumbers = numbers.filter(number => {
+    return number % 2 === 0;
+  });
+  const newNumbers = evenNumbers.map(evenNumber => {
+    return evenNumber * 2;
+  });
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:19]: 👍. You could also explore using method chaining

```
return numbers.filter(number => {
    return number % 2 === 0;
  }).
  map(evenNumber => {
    return evenNumber * 2;
  });
```

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 2]
```diff
@@ -31,8 +31,15 @@ const mondayTasks = [
...
-  // TODO complete this function
+function computeEarnings(tasks, hourlyRate) {
+  const durationTimesWithRate = tasks.map(task => {
+    return task.duration * hourlyRate;
+  });
+  const totalEarningsAsMinutes = durationTimesWithRate.reduce((currentTotal, element) => {
+    return currentTotal + element;
+  }, 0);
+  const totalEarnings = (totalEarningsAsMinutes / 60).toFixed(2);
+  return `€${totalEarnings}`;
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:42]: 👍

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 3]
```diff
@@ -26,26 +26,38 @@ const fruitBasket = [
 ];
 
 // ! Function under test
-function sanitizeFruitBasket(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function sanitizeFruitBasket(fruitBasket, allergicFruit) {
+  const sanitizedFruitBasket = fruitBasket.filter(fruit => {
+    return fruit !== allergicFruit;
+  });
+  return sanitizedFruitBasket;
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:33]: 👍

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js [hunk 4]
```diff
@@ -17,10 +17,12 @@ function createObservable() {
   const subscribers = [];
   return {
     subscribe: function (subscriber) {
-      // TODO complete this function
+      subscribers.push(subscriber);
     },
     notify: function (message) {
-      // TODO complete this function
+      subscribers.forEach(subscriber => {
+        subscriber(message);
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:24]: 👍

**Review by yash-kapila — COMMENTED** [review]
Hi - I've finished reviewing the homework. Can you please have a look at the comments?

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js (hunk 1 above)
Comment (caglar-kaya) [reply 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:19]: Method chaining is also a good way. I just wanted to extract logic for readability and understandability. But method chaining is also clear and simple. Thanks for comment.

**Comment by caglar-kaya** [comment]
Hey @yash-kapila ,
Thank you so much for your review and comments..
Have a nice weekend..

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 5]
```diff
@@ -26,26 +26,38 @@ const fruitBasket = [
...
+    const expected = [
+      'apple',
+      'lemon',
+      'grapefruit',
+      'lemon',
+      'banana',
+      'watermelon',
+      'lemon',
+    ];
+    expect(actual).toEqual(expected);
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:55]: @yash-kapila Should not this test run `sanitizeFruitBasket` function before checking if `actual` is equal to `expected` ?

**Pull request closed without merging**